pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -570,6 +579,66 @@ func UnqualifiedSearchRegistries(ctx *types.SystemContext) ([]string, error) {
 	return config.UnqualifiedSearchRegistries, nil
 }
 
+// GetCredsStore return the list of credetntial store configed by creds_store for registries
+func GetCredsStore(ctx *types.SystemContext) ([]string, error) {
+	config, err := getConfig(ctx)
+	if err != nil {
+		return nil, err
+	}
+	return config.CredsStore, nil
+}
+
+// GetCredConfigs returns a slice of credential stores configred in registries.conf
+// caller should use GetRegistryCredentialHelpers to get cred helper of specified registry
+func GetCredConfigs(sys *types.SystemContext) ([]string, error) {
+	var (
+		credConfigs []string
+		credHelpers = make(map[string]bool)
+	)
+	config, err := getConfig(sys)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to parse the registries.conf file")
+	}
+	for _, store := range config.CredsStore {
+		credHelpers[store] = true
+	}
+	registries, err := GetRegistries(sys)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to parse the registries.conf file")
+	}
+	for _, reg := range registries {
+		if ok, _ := credHelpers[reg.CredHelper]; !ok {
+			credHelpers[reg.CredHelper] = true
+		}
+	}
+	for cred := range credHelpers {
+		credConfigs = append(credConfigs, cred)
+	}
+	return credConfigs, nil
+}
+
+// GetRegistryCredentialHelpers returns the credHelper configured for given registry
+// credHelper configured for the registry can override list of global configuration
+func GetRegistryCredentialHelpers(sys *types.SystemContext, registry string) ([]string, error) {
+	registries, err := GetRegistries(sys)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to parse the registries.conf file")
+	}
+	for _, reg := range registries {
+		if reg.CredHelper != "" && reg.Prefix == registry {
+			return []string{reg.CredHelper}, nil
+		}
+	}
+	config, err := getConfig(sys)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to parse the registries.conf file")
+	}
+	if len(config.CredsStore) > 0 {
+		return config.CredsStore, nil
+	}
+	return nil, nil
```

Comment:
(We might eventually need the `file` fallback here.)

Reply:
Prefer to add  ["file"] configuration in a second PR.  So if there's a "file" in the list, return the original list. If "file" does not exist, append it to the end of the list and can fall back to file when dealing with the credential configs?

Reply:
No — if the user configures _only_ a desktop keyring, we should not be overriding that decision and writing to files. Only if `registries.conf`/`registries.d/*` don’t contain _any_ helper configuration, we would implicitly assume `["file"]`.

(With only this PR, we do always fall back to files, because we first use the helpers, and then always fall back to `readJSON`/`modifyJSON`, but that other PR would change that so that the JSON code only runs if `file` is specified.)

Reply:
I agree with @mtrmac